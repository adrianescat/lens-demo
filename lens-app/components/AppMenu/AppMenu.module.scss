.appMenu {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  padding: 50px;
  flex-wrap: wrap;
  max-width: 1280px;
  margin: 0 auto;
}

.option {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  font-size: 22px;
  border: 1px solid #f0f0f0;
  border-radius: 5px;
  padding: 20px;
  cursor: pointer;
  box-shadow: 0 1px 2px rgba(0,0,0,0.15);
  position: relative;
  margin: 10px;

  &::after {
    content: "";
    position: absolute;
    z-index: -1;
    width: 100%;
    height: 100%;
    opacity: 0;
    top: 0;
    border-radius: 5px;
    box-shadow: 0 5px 15px rgb(0 0 0 / 30%);
    transition: opacity 0.3s ease-in-out;
    left: 0;
  }

  &:hover::after {
    opacity: 1;
  }
}

.unavailable {
  filter: saturate(0%);
}