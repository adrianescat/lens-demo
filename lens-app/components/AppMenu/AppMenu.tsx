import React, { ReactElement } from 'react';
import styles from './AppMenu.module.scss';
import Image from 'next/image';
import Link from 'next/link';

export default function AppMenu(): ReactElement {
  return (
    <section className={styles.appMenu}>
      <Link href="/dashboard">
        <div className={styles.option}>
          <Image
            src="/img/dashboard.svg"
            alt="Go to dashboard"
            width={300}
            height={300}
          />

          <div>DASHBOARD</div>
        </div>
      </Link>

      <div className={styles.option}>
        <Image
          src="/img/add_lens.svg"
          alt="Add Lens"
          width={300}
          height={300}
          className={styles.unavailable}
        />

        <div>ADD LENS</div>
      </div>

      <div className={styles.option}>
        <Image
          src="/img/support.svg"
          alt="Support"
          width={300}
          height={300}
          className={styles.unavailable}
        />

        <div>SUPPORT</div>
      </div>
    </section>
  )
}
