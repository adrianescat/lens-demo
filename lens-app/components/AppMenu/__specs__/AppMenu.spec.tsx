import React from 'react';
import { shallow } from 'enzyme';
import AppMenu from '../AppMenu';

let wrapper: any;
let props: any;

beforeEach(() => {
  props = {}
  wrapper = shallow(<AppMenu {...props} />);
});

test('should match snapshots', () => {
  expect(wrapper).toMatchSnapshot();
});