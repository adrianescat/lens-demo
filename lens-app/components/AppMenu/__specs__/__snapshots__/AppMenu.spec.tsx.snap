// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
<section
  className="appMenu"
>
  <Link
    href="/dashboard"
  >
    <div
      className="option"
    >
      <Image
        alt="Go to dashboard"
        height={300}
        src="/img/dashboard.svg"
        width={300}
      />
      <div>
        DASHBOARD
      </div>
    </div>
  </Link>
  <div
    className="option"
  >
    <Image
      alt="Add Lens"
      className="unavailable"
      height={300}
      src="/img/add_lens.svg"
      width={300}
    />
    <div>
      ADD LENS
    </div>
  </div>
  <div
    className="option"
  >
    <Image
      alt="Support"
      className="unavailable"
      height={300}
      src="/img/support.svg"
      width={300}
    />
    <div>
      SUPPORT
    </div>
  </div>
</section>
`;
